source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -351,6 +351,13 @@ void UdpProxyFilter::ActiveSession::write(const Buffer::Instance& buffer) {
   //       set. We allow the OS to select the right IP based on outbound routing rules if
   //       use_original_src_ip_ is not set, else use downstream peer IP as local IP.
   const Network::Address::Ip* local_ip = use_original_src_ip_ ? addresses_.peer_->ip() : nullptr;
```

Comment:
What happened when use original src ip? Since after `connect()`, the socket will a full 4-tuple associated `{source ip, source port, destination ip, destination port}` that source IP can be the different IP than original src one?

Reply:
The [integration test](https://github.com/envoyproxy/envoy/blob/main/test/extensions/filters/udp/udp_proxy/udp_proxy_integration_test.cc#L286) can guarantee that after `connect()`, `use_original_src_ip` can continue to work.

Reply:
Emm...I didn't see that test set the `use_original_src_ip`. Actually, I didn't find any test in the file set that flag. But correct me if missed something.

Reply:
Yes, you are right. The integration test does not really test `use_original_src_ip`. I found that in the original PR, `use_original_src_ip` is tested in the [unit test](https://github.com/envoyproxy/envoy/pull/12586/files#diff-c73fc65a80218e8cb67b41e20b10ec5a2770e66e67a3714a979eb2dee3f1f281).

Reply:
After manual testing, `use_original_src_ip` is conflict with `connect`. I will escape connecting if `use_original_src_ip` is set.

Reply:
thanks for the update, also could you add a test for it?

Reply:
The [unit test](https://github.com/envoyproxy/envoy/blob/6e31c37f19c3fc5566edc4133380a681c96ab6d9/test/extensions/filters/udp/udp_proxy/udp_proxy_filter_test.cc#L792) can guarantee. BTW, we cannot make integration test since `use_original_src_ip` requires `CAP_NET_ADMIN` or root privilege.

Reply:
great, thanks, my last request is just add comment here explain why use the connect() and why didn't use connect() when use_original_src_ip_

Reply:
Added, thanks.